src/main/java/de/tum/in/www1/artemis/service/ExerciseService.java

```diff
@@ -537,6 +546,66 @@ private void setAssignedTeamIdForExerciseAndUser(Exercise exercise, User user) {
         }
     }
 
+    /**
+     * Gets the exercise statistics by setting values for each field of the <code>CourseManagementOverviewExerciseStatisticsDTO</code>
+     *
+     * @param courseId the id of the course
+     * @param amountOfStudentsInCourse the amount of students in the course
+     * @param exerciseIds the ids of the exercises to get the statistics for
+     * @return A list of filled <code>CourseManagementOverviewExerciseStatisticsDTO</code>
+     */
+    public List<CourseManagementOverviewExerciseStatisticsDTO> getStatisticsForCourseManagementOverview(Long courseId, Integer amountOfStudentsInCourse, List<Long> exerciseIds) {
```

Comment:
the method is rather long, please split it into multiple methods, and add some code comments, so it's clear what you are doing here.